### PR TITLE
Update reduced nav and secondary nav 

### PR DESCRIPTION
**Pull request by minkyngkm**
## Done

- Removed global nav 
- Updated the current nav to be reduced 
- Updated secondary nav

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/
    - Be sure to test on mobile, tablet and desktop screen sizes
- Demo at https://ubuntu-com-11497.demos.haus/

## Issue / Card

Fixes [#5107](https://github.com/canonical-web-and-design/web-squad/issues/5107), [#5072](https://github.com/canonical-web-and-design/web-squad/issues/5072)

## Screenshots
image

## Help

[QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html) - [Commit guidelines](https://canonical-web-and-design.github.io/practices/workflow/commit-messages.html)

### CODE REVIEW

**Comment by minkyngkm** [comment]
Some questions

1. The items move down on medium screens. Should media query breakpoint be adjusted?  Also, the size of reduced nav is always same whether it's mobile or desktop?

image

2. The colour of buttons when it's clicked. Currently it follows vanilla 

image
image

3. When user button is clicked, it overlaps the secondary nav. 
image

**Comment by davegoddard42** [comment]
@minkyngkm - in answer to your questions:

1. Yes, the breakpoint should be changed. There should never be any wrapping on the menu. The height is the same on both devices
2. The colour is fine in the image - keep
3. The overlap is OK for now - that's how it currently works on the live site. When we launch the new mega nav, I will look at the sign in/user menu as part of that.

**Comment by davegoddard42** [comment]
@minkyngkm Overall LGTM, few points:

- When the menu opens, it should slide down from the bottom of the menu, not the top of the screen.
- On the reduced nav, the font should be Ubuntu Light. It looks like that’s not the case? Might be my eyes!
- Possibly out of scope - when the menu is open, the chevron should point upwards (works for reduced nav, but not for the secondary)

Finally, there's a specific user interaction that I think we discussed but isn't well documented. Might be out of scope for what you're doing. When a user goes to a page, such as Cloud, the reduced nav should be off the screen, and only on user scroll up does it appear. See the video below - happy to chat about the operation as it's a weird one.

https://user-images.githubusercontent.com/25453620/164219023-fc316478-c3bf-4ab4-9120-2cadc3f4f01a.mov

**Comment by fitojb** [comment]
Tutorials should have a bubble subnav too, and a bigger heading maybe?:
imatge

**Comment by minkyngkm** [comment]
Thanks for your comment @fitojb. This project is still working in progress but what you pointed out has been updated. Thank you! :)

**Comment by davegoddard42** [comment]
Hi @minkyngkm - some comments below

The URL we're using (/cloud/multi-cloud#secondary-navigation) - is this what will go live? I know we current use /#enterprise for the top nav dropdown, but that's acceptable as the word after the # describes the menu item open. "secondary-navigation" is vague - is there any way around this? 

The menu appears from the bottom of the page - it should appear from the bottom edge of the top menu - I’m not sure if my previous comment was clear. Happy to explain on a call if needed.

The chevron in the menu should rotate not flip - like it does in the current (live) menu (I realise this might have arisen from the video I shared where it doesn’t rotate)  .

When a user lands on a bubble, the reduced top nav should be already hidden - currently there’s an animation where the page slides up.

When a user selects an item from the secondary nav in the bubble, it shouldn’t have any impact on the reduced nav. If the reduced nav is hidden, clicking on another secondary nav item shouldn't show then hide the reduced nav. If the reduced nav is visible, clicking a new secondary nav item shouldn’t hide the reduced nav.

**Comment by anthonydillon** [comment]
@minkyngkm I have switched from using the `secondary-navigation` hash to translating the navigation off-screen until the user scrolls. Also, added a workaround if js is switched off.

@davegoddard42 can you have a look at this demo and see if it is the expected behaviour?

**Comment by davegoddard42** [comment]
Thanks @anthonydillon - I can't see the demo though - it constantly says it's building.

**Comment by anthonydillon** [comment]
@davegoddard42 it's built now

**Comment by carkod** [comment]
@sowasred2012 can you take a look?

File: static/js/src/navigation.js [hunk 1]
```diff
@@ -30,18 +31,51 @@ navDropdowns.forEach(function (dropdown) {
...
+    navDropdowns.forEach(function (dropdown) {
+      if (
+        dropdown.classList.contains("is-selected") &&
+        window.innerWidth < 1024
+      ) {
+        navigation.classList.add("has-menu-open");
+      }
+    });
+  }
+  window.onresize = menuOpenMobile;
```
Comment (anthonydillon) [inline static/js/src/navigation.js:67]: This is likely to produce performance issues. Can we not just use CSS with media queries?

File: static/js/src/navigation.js [hunk 2]
```diff
@@ -226,15 +264,23 @@ if (accountContainer && accountContainerSmall) {
...
         var target = document.getElementById(
           element.getAttribute("aria-controls")
         );
 
         if (show) {
-          container.classList.add("is-selected");
+          container.classList.add("is-active");
+          [].slice.call(dropdownWindow.children).forEach((dropdownContent) => {
+            if (dropdownContent.classList.contains("u-hide") === false) {
+              dropdownContent.classList.add("u-hide");
```
Comment (anthonydillon) [inline static/js/src/navigation.js:279]: class list.add does not dup classes so you don't need the check above

File: templates/templates/_navigation.html [hunk 3]
```diff
@@ -1,107 +1,106 @@
...
+        <a class="p-navigation__link" href="{{ breadcrumbs.section.path }}{% if get_test_backend %}?test_backend=true{% endif %}">
+          <div class="p-navigation__logo-tag">
+            <img class="p-navigation__logo-icon" src="https://assets.ubuntu.com/v1/82818827-CoF_white.svg" alt="">
+          </div>
+          <span class="p-navigation__logo-title">{{ breadcrumbs.section.title | replace(" ", "&nbsp;") | safe }}</span>
+        </a>
+      </div>
+      <div class="u-hide js-account--small"></div>
+      <a href="#secondary-navigation" class="p-navigation__toggle--open" title="menu"><i class="p-icon--chevron-down"></i></a>
+      <a href="#secondary-navigation-closed" class="p-navigation__toggle--close" title="close menu"><i class="p-icon--chevron-up"></i></a>
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:93]: Can you add some text to the `i` to give voice over something to read out

File: templates/templates/_navigation.html [hunk 4]
```diff
@@ -1,107 +1,106 @@
...
+      <div class="p-navigation__tagged-logo">
+        <a class="p-navigation__link" href="{{ breadcrumbs.section.path }}{% if get_test_backend %}?test_backend=true{% endif %}">
+          <div class="p-navigation__logo-tag">
+            <img class="p-navigation__logo-icon" src="https://assets.ubuntu.com/v1/82818827-CoF_white.svg" alt="">
+          </div>
+          <span class="p-navigation__logo-title">{{ breadcrumbs.section.title | replace(" ", "&nbsp;") | safe }}</span>
+        </a>
+      </div>
+      <div class="u-hide js-account--small"></div>
+      <a href="#secondary-navigation" class="p-navigation__toggle--open" title="menu"><i class="p-icon--chevron-down"></i></a>
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:92]: Can you add text to the i

File: templates/templates/_navigation.html [hunk 5]
```diff
@@ -110,126 +109,268 @@ <h5 class="p-navigation--secondary__logo">
...
+  });
 
+  document.addEventListener('scroll', function(e) {
+    if (window.scrollY <= 0) {
+      document.body.style.transform = 'translateY(0px)';
+    }
+  });
+  
   function fetchDropdown(url, id) {
+    var mobile =document.getElementById(id+"-mobile");
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:160]:
```suggestion
    var mobile = document.getElementById(id+"-mobile");
```

File: templates/templates/_navigation.html [hunk 6]
```diff
@@ -110,126 +109,268 @@ <h5 class="p-navigation--secondary__logo">
...
-    dropdownWindowOverlay.classList.remove("fade-animation");
-    navigation.classList.add('has-active-search');
+    var navigation = e.target.closest('.p-navigation');
+    var nav = navigation.querySelector('.p-navigation__nav');
+
+    var searchInput = navigation.querySelector('.p-search-box__input');
+    var buttons = document.querySelectorAll('.js-search-button');
+    var secondaryNav = document.querySelector(".is-secondary")
+    // Replace this one when the secondary items are selected
+    // var secondaryNav = document.querySelector(".is-secondary .p-navigation__banner")
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:223]: Can this be removed?

File: templates/templates/_navigation.html [hunk 7]
```diff
@@ -110,126 +109,268 @@ <h5 class="p-navigation--secondary__logo">
...
+
+    var banner = document.querySelector('.p-navigation__banner');
+    var buttons = document.querySelectorAll('.js-search-button');
+
+    buttons.forEach((searchButton) => {
+      searchButton.removeAttribute('aria-pressed');
+    });
+    var secondaryNav = document.querySelector(".is-secondary")
+    // Replace this one when the secondary items are selected
+    // var secondaryNav = document.querySelector(".is-secondary .p-navigation__banner")
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:264]: Can this be removed?

File: templates/templates/_navigation.html [hunk 8]
```diff
@@ -110,126 +109,268 @@ <h5 class="p-navigation--secondary__logo">
...
     }
   }
-</script>
+}
+
+var navigation = document.querySelector('#navigation');
+initNavigationSearch(navigation);
+
+
+//dropddwon
```
Comment (anthonydillon) [inline templates/templates/_navigation.html:322]: ?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Some comments on the code. It's a huge review so I can't be sure I caught everything. Should this branch be targeting main?

File: static/js/src/navigation.js (hunk 1 above)
Comment (minkyngkm) [reply static/js/src/navigation.js:67]: When the screen gets smaller while the menu is open on big screen, the menu needs to be open on small screen but I couldn't find a way to open the menu with media queries.
ezgif com-gif-maker (5)

**Comment by sowasred2012** [comment]
@marcushas @waynecrosby this is the demo for the new nav on ubuntu.com: https://ubuntu-com-11497.demos.haus/

File: package.json [hunk 9]
```diff
@@ -107,7 +105,7 @@
     "typescript": "4.5.4",
     "url-polyfill": "1.1.12",
     "url-search-params-polyfill": "8.1.1",
-    "vanilla-framework": "3.1.1",
+    "vanilla-framework": "3.2.0",
```
Comment (anthonydillon) [inline package.json:108]: Why did you have to bump the Vanilla version?

File: static/js/src/navigation.js (hunk 1 above)
Comment (anthonydillon) [reply static/js/src/navigation.js:67]: Ok I see the issue. I think this could be improved in the future

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few small niggles.

## QA
- Small white line by the menu on small screens: 
image
- No space between icon and Open: 
image
- When navigating the site the transform on the body does not totally hide the global nav
- Search doesn't work

**Comment by davegoddard42** [comment]
@minkyngkm - some comments below.

1. If I go to an external site (by clicking a link in the mega nav), then navigate back, the menu is in the below state - the last clicked menu item is showing as open in the nav, but there's no menu open. 



2. On mobile, the word "Open" appears next to the down chevron when in a bubble.



3. On mobile, when the bubble menu is opened, the slimline top nav is hidden by an auto scroll up. If it was visible when the user opens the bubble menu, it should stay visible. Opening the bubble nav shouldn’t affect the scroll position. 

4. On mobile, when opening the slimline top nav menu, it covers the open bubble menu. Covering is fine for Desktop, but on mobile it should push the bubble menu down.

**Comment by carkod** [comment]
@anthonydillon @davegoddard42 can you re-review? Thanks

**Review by anthonydillon — APPROVED** [review]
Code and QA LGTM :+1:

**Comment by davegoddard42** [comment]
@minkyngkm Looking awesome - just a couple of final issues, and then that's it!

- When a user switches between tabs in a bubble, this shouldn't affect the scroll status - if the page is scrolled down so the reduced menu is showing, it should stay that way

- When leaving ubuntu.com for an external site, if the user then hits back on their browser, the menu should be closed. Before, the menu was closed but the tab status was "open" - the tab status and menu status are now in line but it should be closed by default.

**Comment by davegoddard42** [comment]
LGTM @minkyngkm @sowasred2012 - thanks!!

**Comment by sowasred2012** [comment]
Marking as QA-1 until I can fix this issue Dave pointed out:



https://user-images.githubusercontent.com/2376968/174116309-d362eca5-8fb6-41cf-bfd5-9a39f997c444.mov

**Comment by davegoddard42** [comment]
LGTM @sowasred2012

**Pull request merged**